packages/astro/src/stores/external.ts

```diff
@@ -124,3 +121,19 @@ export const $signInStore = computed([$clientStore], client => client?.signIn);
  * $signUpStore.subscribe((signUp) => console.log(signUp.status))
  */
 export const $signUpStore = computed([$clientStore], client => client?.signUp);
+
+/**
+ * Records a telemetry event when a store is used to match React hooks telemetry.
+ *
+ * @param {Store} store - The nanostore instance to monitor.
+ * @param {string} method - The name of the method associated with the store usage.
+ */
+const recordTelemetryEvent = (store: Store, method: string) => {
+  onMount(store, () => {
+    $clerk.get()?.telemetry?.record(eventMethodCalled(method));
+  });
+};
+
+recordTelemetryEvent($signInStore, '$signInStore');
+recordTelemetryEvent($signUpStore, '$signUpStore');
+recordTelemetryEvent($organizationStore, '$organizationStore');
```

Comment:
We're still unsure about this. Do we have any criteria on recording telemetry events in the client?

This nanostores are basically 1:1 of React hooks. When a store is used, the `onMount` callback gets invoked and that's the time we record the event

Reply:
it's kind of tough to find a common solution for hooks and stores here. I think we can continue with this, and then later in analytics calculate the actual usage of stores by subtracting the usage of its respective hook.

something among the lines of 
```ts
userStoreUsage = getCount(`$userStore`) - `getCount(`$useUser`)
```

does this make sense to you ? Do you have an alternative ?

Also we can always use new computed stores for our hooks in order to have a different reference.

Reply:
Yeah it makes sense, so that if we update later we can track the exact usage of stores (not the wrapping hooks). Let's go with it then for now